### PR TITLE
Add a more parallel version of chameneos

### PR DESCRIPTION
**Pull request by ronawho**
Add a version of chameneous that runs the population sizes in parallel instead
of running them one after another. This is what the fastest gcc/g++/go versions
do. It requires saving the initial colors since they still have to printed out
in order.

This has a decent performance impact taking execution time from ~4.7 seconds to
~/3.8 seconds on the shootout-like box.

### CODE REVIEW

File: test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl [hunk 1]
```diff
@@ -0,0 +1,291 @@
+/* The Computer Language Benchmarks Game
+   http://benchmarksgame.alioth.debian.org/
+
+   contributed by Hannah Hemmaplardh, Lydia Duncan, and Brad Chamberlain
```
Comment (lydia-duncan) [inline test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl]: You should add your name to the list of contributors

File: test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl [hunk 2]
```diff
@@ -0,0 +1,291 @@
...
+
+  const group1 = new Population(popSize1);
+  const group2 = new Population(popSize2);
+
+  cobegin {
+    group1.holdMeetings(n);
+    group2.holdMeetings(n);
+  }
+
+  group1.printColors();
```
Comment (bradcray) [inline test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl]: Elliot, do you understand how printing the colors after the meetings is generating the same result as our previous version which prints the colors before holding the meetings?  I can't figure out what I must be missing...

Comment (bradcray) [inline test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl]: Oh, they store their initial color...  pretty sneaky sis (I should've read the PR message, obviously).  Was there a downside to printing the colors first?

Comment (ronawho) [inline test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl]: Mentioned this in person, but for the record:

The [expected output](http://benchmarksgame.alioth.debian.org/download/chameneosredux-output.txt) is
```
  // group1 initial colors

  // group1 output

  // group2 initial colors

  // group2 output 
```

```
- write the colours the creatures start with
- after N meetings have taken place, for each creature write the number of creatures met and spell out the number of times the creature met a creature with the same name (should be zero)
```
so you can't just print the colors before holding meetings. Technically you could print group 1's meetings before the cobegin, but that looks strange IMO

Comment (bradcray) [inline test/studies/shootout/chameneos-redux/elliot/chameneosredux-ejr.chpl]: @ronawho figured out the answer: We must print out all of group1's stuff before any of group2's so if they're meeting simultaneously, we have to preserve the original colors.

**Pull request merged**